orc8r/gateway/python/magma/magmad/metrics_collector.py

```diff
@@ -151,6 +175,76 @@ def _add_uptime_metric(self, service_name, family):
         if uptime is not None:
             self._samples.append(uptime)
 
+    def scrape_prometheus_target(self, target: ScrapeTarget) -> None:
```

Comment:
minor nit: We could probably move this to async await code instead of futures and callback.

Reply:
Seems like a bigger BE project for all of Magmad since everything is written like this.